docs/architecture/adr-102-rpc-companion.md

```diff
@@ -0,0 +1,341 @@
+| `status`               |                       | Gets node status                          | The request should proxy to the full node since the RPC companion database will not store node status information                                                                                                                               | yes   |
+
+> NOTE: The RPC Companion should not implement logic to store data in its database that can modify state in the blockchain such as
+the `broadcast_tx_*` methods.  These requests will proxy to the full node as outlined above.
+
+### High-level architecture
+
+![High-level architecture](images/adr-102-architecture.jpeg)
+
+This diagram shows all the required components for a full RPC Companion solution. The solution implementation contains
+many parts and each one is described below:
+
+### [Ingest Service](#ingest-service)
+
+The **Ingest Service** pulls the data from the full node RPC endpoint and stores the information retrieved in
+the RPC Companion database.
+
+> In the future, if a gRPC interface is implemented in the full node this might be used to pull the data
+from the server.
+
+The **Ingest Service** can control the pruning on the full node via a mechanism to [pruning service](https://github.com/cometbft/cometbft/blob/thane/adr-084-data-companion-pull-api/docs/architecture/adr-101-data-companion-pull-api.md#pruning-service).
+Once the Ingest Service pulls the data from the full node and is able to process it, and it gets an acknowledgement from the database that the data was inserted,
+the **Ingest Service** can communicate with the full node notifying it that a specific height has been processed and set the processed
+height as the `retain height` on the full node signaling this way to the node that this height can be pruned.
+
+If the **Ingest Service** becomes unavailable (e.g. stops), then it should resume synchronization with the full node when it is back online.
+The **Ingest Service** should query the full node for the last `retain height` and the **Ingest Service** should request
+and process all the heights missing on the database until it catches up with the full node latest height.
+
+In case the **Ingest Service** becomes unavailable for a long time and there are several heights to be synchronized, it is
+important for the **Ingest Service** to do it in a throttled way in order not to stress the full node and cause issues in its consensus processing.
+
+### [Database](#database)
+
+The database stores the data retrieved from the full node and provide this data for the RPC server instance. Since the frequency
+that blocks are generated on the chain are in the range from 5 seconds to 7 seconds on average, the _write_ back pressure is not
```

Comment:
This data (5-7 seconds) is from which network? I would make this affirmation less network-dependent, while providing this data (5-7s) as an example of a specific network. Moreover, this is the _period_ not the _frequency_.

For instance, one can say that the frequency is not more than 2 blocks per second, which would lead to 0.5s period, and more likely in the order of 8-12 blocks per minute?

Reply:
makes sense, thanks, re-wrote it.